File: src/main/java/com/kmarzecki/communicator/repository/RoleRepository.java
package com.kmarzecki.communicator.repository;

import com.kmarzecki.communicator.model.auth.Role;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * Role repository
 */
public interface RoleRepository extends JpaRepository<Role, Integer> {
    /**
     * find role by name
     * @param roleName role name
     * @return Role
     */
    Role findByName(String roleName);
}

File: src/main/kotlin/com/kmarzecki/trollo/repository/RoleRepository.kt
package com.kmarzecki.trollo.repository

import com.kmarzecki.trollo.model.auth.Role
import org.springframework.data.jpa.repository.JpaRepository

/**
 * Role repository
 */
interface RoleRepository : JpaRepository<Role?, Int?> {
    fun findByName(roleName: String?): Role?
}